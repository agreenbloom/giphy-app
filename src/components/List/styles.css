.imageContainer {
  border: 1px solid black;
  padding: 10px 10px 0;
  border-radius: 10px;
  text-decoration: none;
  color: black;
  font-size: 14px;
  position: relative;
}

.gallery {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  margin: 20px;
  align-items: center;
  grid-gap: 10px;
  text-align: center;
  padding: 20px;
  max-width: 1000px;
  margin: 0 auto;
}

.gallery img {
  max-width: 100%;
  width: 100%;
}


@keyframes fadeInRight {
  from {
    opacity: 0;
    transform: translateX(100%);
  }

  to {
    opacity: 1;
    transform: translateX(0%);
  }
}

.ratingContainer {
  position: absolute;
  right: -2500px;
  display: flex;
  align-items: center;
}

.showItemsHover {
  right: 0;
    overflow-x: visible;
    opacity: 1;
    -webkit-animation: fadeInRight 0.2s;
    animation: fadeInRight 0.2s;
    -webkit-animation-fill-mode: both;
    animation-fill-mode: both;
    bottom: 0;
    left: 0;
    background: rgba(255, 255, 255, 0.86);
    height: 24px;
    padding: 5px 20px;
}
