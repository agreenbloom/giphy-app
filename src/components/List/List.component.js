import React, {Component} from 'react'
import PropTypes from 'prop-types';

import Rating from '../Rating/Rating.component.js';
import Lightbox from 'react-images';

import "./styles.css";

export default class List extends Component {
  static propTypes = {
    disabled: PropTypes.bool,
    handleRatingClick: PropTypes.func,
  };

  constructor () {
		super();

		this.state = {
			lightboxIsOpen: false,
			currentImage: 0,
		};
  }


  openLightbox = (index, event) => {
		event.preventDefault();
		this.setState({
			currentImage: index,
			lightboxIsOpen: true,
		});
	}

	closeLightbox = () => {
		this.setState({
			currentImage: 0,
			lightboxIsOpen: false,
		});
	}

	gotoPrevious = () => {
    const currentImage = this.state.currentImage - 1;
		this.setState({
			currentImage,
		});
	}

	gotoNext = () => {
    const currentImage = this.state.currentImage + 1;
		this.setState({
			currentImage,
		});
	}

	handleClickImage = () => {
		if (this.state.currentImage === this.props.gifs.length - 1) return;
		this.gotoNext();
	}

  handleRatingClick = (i, rating ) => {
    let currentImage = this.props.gifs[i];
    currentImage["rating"] = rating;

    if(this.props.handleRatingClick) this.props.handleRatingClick(currentImage)
  }

  renderGallery = () => {
		const { gifs } = this.props;

		if (!gifs) return;

		const gallery = gifs.map((img, i) => {
			return (
        <div className="imageContainer" key={i}>
  				<a
  					href={img.preview}
  					key={i}
  					onClick={(e) => this.openLightbox(i, e)}
            className="card"
  				>
  					<img src={img.preview}  alt=''/>
  				</a>
          <div className="ratingContainer">
            {console.log('img.ra', img.rating)}
            <Rating starsSelected={img.rating ? img.rating : 0} handleOnClick={this.handleRatingClick.bind(this, i)}/>
          </div>
        </div>
			);
		});

		return (
			<div className={'gallery'}>
				{gallery}
			</div>
		);
	}

  render() {
    const { gifs } = this.props;
    const { currentImage, lightboxIsOpen } = this.state;

    return(
      <React.Fragment>
        {this.renderGallery()}
        <Lightbox
					currentImage={currentImage}
					images={gifs}
					isOpen={lightboxIsOpen}
					onClickImage={this.handleClickImage}
					onClickNext={this.gotoNext}
					onClickPrev={this.gotoPrevious}
					onClose={this.closeLightbox}

				/>
      </React.Fragment>
    )
  }
}
