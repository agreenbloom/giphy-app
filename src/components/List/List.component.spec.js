import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.component';
import { shallow } from 'enzyme';

const mockGifs = [
  {
    caption: "happy very funny GIF by Disney Zootopia",
    id:"3NtY188QaxDdC",
    preview: "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
    rating: 4,
    slug: "3NtY188QaxDdC",
    src: "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
    timeTrending: "2018-11-19 23:00:02",
  }, {
    caption:"bird no GIF by Cheezburger",
    id: "g69ZPJfLy7hD2",
    preview: "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
    rating: 5,
    slug: "cheezburger-no-bird-g69ZPJfLy7hD2",
    src: "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
    timeTrending: "2017-08-19 22:45:01",
  }
];

it('renders without crashing', () => {
  const wrapper = shallow(<List gifs={mockGifs} />);

  expect(wrapper).toMatchSnapshot();
});


describe('methods', () => {

  describe('closeLightbox', () => {
    it('should close lightbox', () => {

      const wrapper = shallow(<List gifs={mockGifs}  />);
      wrapper.instance().closeLightbox()
      expect(wrapper.instance().state.lightboxIsOpen).toBe(false);
    });
  });

  describe('gotoPrevious', () => {
    it('should close lightbox', () => {
      const mockFunction = jest.fn()
      const wrapper = shallow(<List gifs={mockGifs} handleRatingClick={mockFunction} />);
      wrapper.setState({
        currentImage: 1
      })
      wrapper.instance().gotoPrevious();
      expect(wrapper.instance().state.currentImage).toBe(0);
    });
  });


  describe('handleRatingImage', () => {
    it('should close lightbox', () => {
      const mockFunction = jest.fn()
      const wrapper = shallow(<List gifs={mockGifs} handleRatingClick={mockFunction} />);
      wrapper.instance().handleRatingClick(1, 5);

      expect(mockFunction).toHaveBeenCalled();
    });
  });

});
