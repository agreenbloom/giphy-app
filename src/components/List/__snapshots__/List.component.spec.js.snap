// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    gifs={
      Array [
        Object {
          "caption": "happy very funny GIF by Disney Zootopia",
          "id": "3NtY188QaxDdC",
          "preview": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
          "rating": 4,
          "slug": "3NtY188QaxDdC",
          "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
          "timeTrending": "2018-11-19 23:00:02",
        },
        Object {
          "caption": "bird no GIF by Cheezburger",
          "id": "g69ZPJfLy7hD2",
          "preview": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
          "rating": 5,
          "slug": "cheezburger-no-bird-g69ZPJfLy7hD2",
          "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
          "timeTrending": "2017-08-19 22:45:01",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="gallery"
        >
          <div
            className="imageContainer"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              className="card"
              href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
              onClick={[Function]}
            >
              <img
                alt=""
                src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
              />
            </a>
            <div
              className="ratingContainer"
            >
              <Rating
                handleOnClick={[Function]}
                starsSelected={4}
                totalStars={5}
              />
            </div>
          </div>
          <div
            className="imageContainer"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              className="card"
              href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
              onClick={[Function]}
            >
              <img
                alt=""
                src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
              />
            </a>
            <div
              className="ratingContainer"
            >
              <Rating
                handleOnClick={[Function]}
                starsSelected={5}
                totalStars={5}
              />
            </div>
          </div>
        </div>,
        <Lightbox
          closeButtonTitle="Close (Esc)"
          currentImage={0}
          enableKeyboardInput={true}
          imageCountSeparator=" of "
          images={
            Array [
              Object {
                "caption": "happy very funny GIF by Disney Zootopia",
                "id": "3NtY188QaxDdC",
                "preview": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                "rating": 4,
                "slug": "3NtY188QaxDdC",
                "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
                "timeTrending": "2018-11-19 23:00:02",
              },
              Object {
                "caption": "bird no GIF by Cheezburger",
                "id": "g69ZPJfLy7hD2",
                "preview": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                "rating": 5,
                "slug": "cheezburger-no-bird-g69ZPJfLy7hD2",
                "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
                "timeTrending": "2017-08-19 22:45:01",
              },
            ]
          }
          isOpen={false}
          leftArrowTitle="Previous (Left arrow key)"
          onClickImage={[Function]}
          onClickNext={[Function]}
          onClickPrev={[Function]}
          onClickShowNextImage={true}
          onClose={[Function]}
          preloadNextImage={true}
          preventScroll={true}
          rightArrowTitle="Next (Right arrow key)"
          showCloseButton={true}
          showImageCount={true}
          spinner={[Function]}
          spinnerColor="white"
          spinnerSize={100}
          theme={Object {}}
          thumbnailOffset={2}
          width={1024}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="imageContainer"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="card"
                href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                onClick={[Function]}
              >
                <img
                  alt=""
                  src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                />
              </a>
              <div
                className="ratingContainer"
              >
                <Rating
                  handleOnClick={[Function]}
                  starsSelected={4}
                  totalStars={5}
                />
              </div>
            </div>,
            <div
              className="imageContainer"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="card"
                href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                onClick={[Function]}
              >
                <img
                  alt=""
                  src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                />
              </a>
              <div
                className="ratingContainer"
              >
                <Rating
                  handleOnClick={[Function]}
                  starsSelected={5}
                  totalStars={5}
                />
              </div>
            </div>,
          ],
          "className": "gallery",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="card"
                  href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                  />
                </a>,
                <div
                  className="ratingContainer"
                >
                  <Rating
                    handleOnClick={[Function]}
                    starsSelected={4}
                    totalStars={5}
                  />
                </div>,
              ],
              "className": "imageContainer",
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                  />,
                  "className": "card",
                  "href": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Rating
                    handleOnClick={[Function]}
                    starsSelected={4}
                    totalStars={5}
                  />,
                  "className": "ratingContainer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "handleOnClick": [Function],
                    "starsSelected": 4,
                    "totalStars": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="card"
                  href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                  />
                </a>,
                <div
                  className="ratingContainer"
                >
                  <Rating
                    handleOnClick={[Function]}
                    starsSelected={5}
                    totalStars={5}
                  />
                </div>,
              ],
              "className": "imageContainer",
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                  />,
                  "className": "card",
                  "href": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Rating
                    handleOnClick={[Function]}
                    starsSelected={5}
                    totalStars={5}
                  />,
                  "className": "ratingContainer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "handleOnClick": [Function],
                    "starsSelected": 5,
                    "totalStars": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeButtonTitle": "Close (Esc)",
          "currentImage": 0,
          "enableKeyboardInput": true,
          "imageCountSeparator": " of ",
          "images": Array [
            Object {
              "caption": "happy very funny GIF by Disney Zootopia",
              "id": "3NtY188QaxDdC",
              "preview": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
              "rating": 4,
              "slug": "3NtY188QaxDdC",
              "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
              "timeTrending": "2018-11-19 23:00:02",
            },
            Object {
              "caption": "bird no GIF by Cheezburger",
              "id": "g69ZPJfLy7hD2",
              "preview": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
              "rating": 5,
              "slug": "cheezburger-no-bird-g69ZPJfLy7hD2",
              "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
              "timeTrending": "2017-08-19 22:45:01",
            },
          ],
          "isOpen": false,
          "leftArrowTitle": "Previous (Left arrow key)",
          "onClickImage": [Function],
          "onClickNext": [Function],
          "onClickPrev": [Function],
          "onClickShowNextImage": true,
          "onClose": [Function],
          "preloadNextImage": true,
          "preventScroll": true,
          "rightArrowTitle": "Next (Right arrow key)",
          "showCloseButton": true,
          "showImageCount": true,
          "spinner": [Function],
          "spinnerColor": "white",
          "spinnerSize": 100,
          "theme": Object {},
          "thumbnailOffset": 2,
          "width": 1024,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="gallery"
          >
            <div
              className="imageContainer"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="card"
                href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                onClick={[Function]}
              >
                <img
                  alt=""
                  src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                />
              </a>
              <div
                className="ratingContainer"
              >
                <Rating
                  handleOnClick={[Function]}
                  starsSelected={4}
                  totalStars={5}
                />
              </div>
            </div>
            <div
              className="imageContainer"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="card"
                href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                onClick={[Function]}
              >
                <img
                  alt=""
                  src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                />
              </a>
              <div
                className="ratingContainer"
              >
                <Rating
                  handleOnClick={[Function]}
                  starsSelected={5}
                  totalStars={5}
                />
              </div>
            </div>
          </div>,
          <Lightbox
            closeButtonTitle="Close (Esc)"
            currentImage={0}
            enableKeyboardInput={true}
            imageCountSeparator=" of "
            images={
              Array [
                Object {
                  "caption": "happy very funny GIF by Disney Zootopia",
                  "id": "3NtY188QaxDdC",
                  "preview": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                  "rating": 4,
                  "slug": "3NtY188QaxDdC",
                  "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
                  "timeTrending": "2018-11-19 23:00:02",
                },
                Object {
                  "caption": "bird no GIF by Cheezburger",
                  "id": "g69ZPJfLy7hD2",
                  "preview": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                  "rating": 5,
                  "slug": "cheezburger-no-bird-g69ZPJfLy7hD2",
                  "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
                  "timeTrending": "2017-08-19 22:45:01",
                },
              ]
            }
            isOpen={false}
            leftArrowTitle="Previous (Left arrow key)"
            onClickImage={[Function]}
            onClickNext={[Function]}
            onClickPrev={[Function]}
            onClickShowNextImage={true}
            onClose={[Function]}
            preloadNextImage={true}
            preventScroll={true}
            rightArrowTitle="Next (Right arrow key)"
            showCloseButton={true}
            showImageCount={true}
            spinner={[Function]}
            spinnerColor="white"
            spinnerSize={100}
            theme={Object {}}
            thumbnailOffset={2}
            width={1024}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="imageContainer"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <a
                  className="card"
                  href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                  />
                </a>
                <div
                  className="ratingContainer"
                >
                  <Rating
                    handleOnClick={[Function]}
                    starsSelected={4}
                    totalStars={5}
                  />
                </div>
              </div>,
              <div
                className="imageContainer"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <a
                  className="card"
                  href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                  />
                </a>
                <div
                  className="ratingContainer"
                >
                  <Rating
                    handleOnClick={[Function]}
                    starsSelected={5}
                    totalStars={5}
                  />
                </div>
              </div>,
            ],
            "className": "gallery",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="card"
                    href="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                    />
                  </a>,
                  <div
                    className="ratingContainer"
                  >
                    <Rating
                      handleOnClick={[Function]}
                      starsSelected={4}
                      totalStars={5}
                    />
                  </div>,
                ],
                "className": "imageContainer",
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      src="https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif"
                    />,
                    "className": "card",
                    "href": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Rating
                      handleOnClick={[Function]}
                      starsSelected={4}
                      totalStars={5}
                    />,
                    "className": "ratingContainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "handleOnClick": [Function],
                      "starsSelected": 4,
                      "totalStars": 5,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="card"
                    href="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                    />
                  </a>,
                  <div
                    className="ratingContainer"
                  >
                    <Rating
                      handleOnClick={[Function]}
                      starsSelected={5}
                      totalStars={5}
                    />
                  </div>,
                ],
                "className": "imageContainer",
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      src="https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif"
                    />,
                    "className": "card",
                    "href": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Rating
                      handleOnClick={[Function]}
                      starsSelected={5}
                      totalStars={5}
                    />,
                    "className": "ratingContainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "handleOnClick": [Function],
                      "starsSelected": 5,
                      "totalStars": 5,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeButtonTitle": "Close (Esc)",
            "currentImage": 0,
            "enableKeyboardInput": true,
            "imageCountSeparator": " of ",
            "images": Array [
              Object {
                "caption": "happy very funny GIF by Disney Zootopia",
                "id": "3NtY188QaxDdC",
                "preview": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy-preview.gif",
                "rating": 4,
                "slug": "3NtY188QaxDdC",
                "src": "https://media3.giphy.com/media/3NtY188QaxDdC/giphy.gif",
                "timeTrending": "2018-11-19 23:00:02",
              },
              Object {
                "caption": "bird no GIF by Cheezburger",
                "id": "g69ZPJfLy7hD2",
                "preview": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy-preview.gif",
                "rating": 5,
                "slug": "cheezburger-no-bird-g69ZPJfLy7hD2",
                "src": "https://media3.giphy.com/media/g69ZPJfLy7hD2/giphy.gif",
                "timeTrending": "2017-08-19 22:45:01",
              },
            ],
            "isOpen": false,
            "leftArrowTitle": "Previous (Left arrow key)",
            "onClickImage": [Function],
            "onClickNext": [Function],
            "onClickPrev": [Function],
            "onClickShowNextImage": true,
            "onClose": [Function],
            "preloadNextImage": true,
            "preventScroll": true,
            "rightArrowTitle": "Next (Right arrow key)",
            "showCloseButton": true,
            "showImageCount": true,
            "spinner": [Function],
            "spinnerColor": "white",
            "spinnerSize": 100,
            "theme": Object {},
            "thumbnailOffset": 2,
            "width": 1024,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
